vcd/datasource_vcd_nsxt_alb_virtual_service_sec_rules.go

```diff
@@ -0,0 +1,193 @@
+package vcd
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func datasourceVcdAlbVirtualServiceSecRules() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: datasourceVcdAlbVirtualServiceSecRulesRead,
+
+		Schema: map[string]*schema.Schema{
```

Comment:
I see the three data sources schemas are pretty similar, is it possible for them to share the common pieces of the schema? Like
```suggestion
		Schema: dsVcdAlbVirtualServiceGenericRules("security")
		Schema: dsVcdAlbVirtualServiceGenericRules("request")
		Schema: dsVcdAlbVirtualServiceGenericRules("response")
```

and the function shares the common parts and decides about different ones.

Or maybe it would be too confusing when reading (due too many nested stuff)?

Reply:
Yeah, so I also had this in mind, but although it might be a bit more DRY, following schemas could get completely out of hand. It is quite messy already. Not very strong against it if there are your upvotes for it, but I suspect this won't aid much :)

I was also thinking similarly about the CRUD functions for all 3 resources. But then again, some conditions within the code. And if for whatever reason they need to diverge - one has to split them at first